server/user.go

```diff
@@ -0,0 +1,345 @@
+               </script>
+       </head>
+       <body>
+               <p>Disconnected from JIRA. Please close this page.</p>
+       </body>
+</html>
+`
+
+	w.Header().Set("Content-Type", "text/html")
+	w.Write([]byte(html))
+
+	return http.StatusOK, nil
+}
+
+func (p *Plugin) handleHTTPUserConfig(w http.ResponseWriter, r *http.Request) (int, error) {
+	ji, err := p.LoadCurrentJIRAInstance()
+	if err != nil {
+		return http.StatusInternalServerError, err
+	}
+
+	_, tokenString, err := validateJWT(r, ji.asc)
+	if err != nil {
+		return http.StatusBadRequest, err
+	}
+
+	// TODO: Ideally find a way to display a message in the form that includes
+	// the MM user ID, not yet sure how to best do it.
+
+	bb := &bytes.Buffer{}
+	err = p.userConfigTemplate.ExecuteTemplate(bb, "config",
+		struct {
+			JWT        string
+			ArgMMToken string
+		}{
+			JWT:        tokenString,
+			ArgMMToken: argMMToken,
+		})
+	if err != nil {
+		return http.StatusInternalServerError, err
+	}
+	w.Header().Set("Content-Type", "text/html")
+	io.Copy(w, bytes.NewReader(bb.Bytes()))
+	return http.StatusOK, nil
+}
+
+func (p *Plugin) handleHTTPUserConfigSubmit(w http.ResponseWriter, r *http.Request) (int, error) {
+	ji, err := p.LoadCurrentJIRAInstance()
+	if err != nil {
+		return http.StatusInternalServerError, err
+	}
+
+	jwtToken, _, err := validateJWT(r, ji.asc)
+	if err != nil {
+		return http.StatusBadRequest, err
+	}
+	claims, ok := jwtToken.Claims.(jwt.MapClaims)
+	if !ok {
+		return http.StatusBadRequest, fmt.Errorf("invalid JWT claims")
+	}
+	context, ok := claims["context"].(map[string]interface{})
+	if !ok {
+		return http.StatusBadRequest, fmt.Errorf("invalid JWT claim context")
+	}
+	user, ok := context["user"].(map[string]interface{})
+	if !ok {
+		return http.StatusBadRequest, fmt.Errorf("invalid JWT: no user data")
+	}
+
+	userKey, _ := user["userKey"].(string)
+	username, _ := user["username"].(string)
+	accountId, _ := user["accountId"].(string)
+	displayName, _ := user["displayName"].(string)
+
+	mmToken := r.Form.Get(argMMToken)
+	uinfo := JIRAUserInfo{
+		Key:       userKey,
+		AccountId: accountId,
+		Name:      username,
+	}
+
+	mattermostUserId, err := p.ParseAuthToken(mmToken)
+	if err != nil {
+		return http.StatusBadRequest, err
+	}
+
+	err = p.StoreUserInfo(ji, mattermostUserId, uinfo)
+	if err != nil {
+		return http.StatusInternalServerError, err
+	}
+
+	p.API.PublishWebSocketEvent(
+		WS_EVENT_CONNECT,
+		map[string]interface{}{
+			"is_connected":    true,
+			"jira_username":   uinfo.Name,
+			"jira_account_id": uinfo.AccountId,
+			"jira_url":        ji.asc.BaseURL,
+		},
+		&model.WebsocketBroadcast{UserId: mattermostUserId},
+	)
+
+	mmuser, aerr := p.API.GetUser(mattermostUserId)
+	if aerr != nil {
+		return http.StatusInternalServerError, aerr
+	}
+	// <script src="https://connect-cdn.atl-paas.net/all.js" data-options="base:true" async></script>
+	w.Header().Set("Content-Type", "text/html")
+	w.Write([]byte(`<!DOCTYPE html>
```

Comment:
Why does this return some HTML. Does the user ever see this? Shouldn't this redirect the user back to MM?

Reply:
This is TBD later